lib/puppet/network/http/connection.rb

```diff
@@ -200,6 +231,36 @@ def request_with_redirects(request, options)
       raise RedirectionLimitExceededException, _("Too many HTTP redirections for %{host}:%{port}") % { host: @host, port: @port }
     end
 
+    # Parse the value of a Retry-After header
+    #
+    # Parses a string containing an Integer or RFC 2822 datestamp and returns
+    # an integer number of seconds before a request can be retried.
+    #
+    # @param header_value [String] The value of the Retry-After header.
+    #
+    # @return [Integer] Number of seconds to wait before retrying the
+    #   request.
+    # @return [nil] Returns `nil` when the `header_value` can't be
+    #   parsed as an Integer or RFC 2822 date.
+    def parse_retry_after_header(header_value)
+      retry_after = begin
```

Comment:
A bit of code golf here to simplify, but maybe this?

```
begin
  Integer(header_value)
rescue TypeError, ArgumentError
  begin
    (DateTime.rfc2822(header_value).to_time - DateTime.now.to_time).to_i
  rescue ArgumentError
    nil
  end
end
```
I don't see a need for the `retry_after` variable, nor for doing this in two parts.

Reply:
The main reason for splitting this into two parts is to ensure we're only rescuing `ArgumentError` when thrown by the `DateTime.rfc2822` parsing method as that is the only operation where failure would be expected.